backend/api/tests/contract-tests/tests/testClientRegistryContract.spec.ts

```diff
@@ -73,61 +23,121 @@ describe("ClientRegistryProvider", () => {
     });
 
     describe("When a POST request is made to create a client", () => {
-        test("it should return a message", async () => {
+        const postEvent: RegisterClientPayload = {
+            contactEmail: "pacttest.account@digital.cabinet-office.gov.uk",
+            service: {
+                serviceName: "My test service",
+                id: "service#testRandomId"
+            }
+        };
+
+        const client_config = {
+            client_name: postEvent.service.serviceName,
+            public_key: public_key,
+            redirect_uris: ["http://localhost/"],
+            contacts: [postEvent.contactEmail],
+            scopes: ["openid", "email", "phone"],
+            subject_type: "pairwise",
+            service_type: "MANDATORY",
+            sector_identifier_uri: "http://gov.uk",
+            client_locs: ["P2"]
+        };
+    
+        const { client_locs, ...expected_returned_values } = client_config;
+
+        test("it should return an a client registry object when adding a client", async () => {
             provider
-                .uponReceiving("add a Client")
+                .uponReceiving("configuration to add a client")
                 .withRequest({
                     method: "POST",
                     path: "/connect/register",
                     contentType: "application/json",
                     headers: {
-                        "Content-Type": "application/json; charset=utf-8"
+                        "Content-Type": "application/json"
                     },
-                    body: EXPECTED_BODY
+                    body: JSON.stringify(client_config)
                 })
                 .willRespondWith({
                     status: 200,
-                    body: eachLike(EXPECTED_BODY),
+                    body: eachLike(expected_returned_values),
                     contentType: "application/json",
                     headers: {
                         "Content-Type": "application/json; charset=utf-8"
                     }
                 });
 
             await provider.executeTest(async mockProvider => {
-                const [results] = await Promise.all([postRequest(mockProvider.url, EXPECTED_BODY)]);
-                console.log("RESULTS: " + JSON.stringify(results));
-                expect(results).toContain(EXPECTED_BODY);
+                process.env.AUTH_REGISTRATION_BASE_URL = mockProvider.url;
+                const results = await registerClientHandler(postEvent, mockLambdaContext);
+                expect(JSON.parse(results.body)).toMatchObject(expected_returned_values);
```

Comment:
What is more important to check in here. That the response body has the values that is sent or it is a client registry object.

Reply:
Hmm I think we should match the same level of validation currently in the code - it's quite permissive right now. If we start asserting the values a bit more - these tests can be updated

Reply:
The current code is quite permissive, so I think we should match that level of assertion. If we do more validation in the handler we should update these tests!